### ISSUE TITLE
alsaloop bails after a while

### ISSUE THREAD

**Issue by rtertiaer** [micro-nova/AmpliPi#663]
This happened post #570 , but that may be unrelated; it may also be an upgrade of system software during the upgrade process that caused this. If I recall correctly, @linknum23 saw this behavior after first testing that PR.

System log:
```
Mar 08 10:09:58 amplipi-inhouse authbind[826]: Pianobar config at /home/pi/amplipi-dev/config/srcs/v1/.config/pianobar
Mar 08 10:09:58 amplipi-inhouse authbind[826]: Activating chicago house, by way of gene farris (temporarily)
Mar 08 10:09:58 amplipi-inhouse authbind[826]:   starting connection via: /home/pi/amplipi-dev/venv/bin/python /home/pi/amplipi-dev
/streams/process_monitor.py alsaloop -C lb1p -P ch3 -t 100000
[snip]
Mar 08 10:17:59 amplipi-inhouse authbind[826]: alsaloop: pcm.c:1654: snd_pcm_poll_descriptors_count: Assertion `pcm' failed.
Mar 08 10:17:59 amplipi-inhouse authbind[826]: Starting process monitor for process: ['alsaloop', '-C', 'lb1p', '-P', 'ch3', '-t', '100000']
Mar 08 10:17:59 amplipi-inhouse authbind[826]: Subprocess exited with non-zero code: -6. Restarting...
[snip]
Mar 08 10:25:00 amplipi-inhouse authbind[826]: alsaloop: pcm.c:1654: snd_pcm_poll_descriptors_count: Assertion `pcm' failed.
Mar 08 10:25:00 amplipi-inhouse authbind[826]: Subprocess exited with non-zero code: -6. Restarting...
[snip]
Mar 08 10:31:19 amplipi-inhouse authbind[826]: alsaloop: pcm.c:1654: snd_pcm_poll_descriptors_count: Assertion `pcm' failed.
Mar 08 10:31:19 amplipi-inhouse authbind[826]: Subprocess exited with non-zero code: -6. Restarting...
```

Kernel log shows nothing.

This has consistently happened with Pandora in particular, though I suppose I would be a bit surprised if that was related.

**Comment by rtertiaer** [comment]
I've seen our airplay setup handle this restarting process fine now; the bug reported here might be limited to Pandora/pianobar not properly bailing or handling a failure case.

**Comment by rtertiaer** [comment]
We've had a couple more reports of Pandora bailing in the field at this point. I'm running a revert branch https://github.com/micro-nova/AmpliPi/tree/pianobar_no_procmon on our in-house unit now to see if it was the work we did to put `pianobar` into a process monitor that is somehow causing this.

**Comment by rtertiaer** [comment]
I believe this is fixed with #688. We can continue to monitor for this but this probably ought to be closed for now.
